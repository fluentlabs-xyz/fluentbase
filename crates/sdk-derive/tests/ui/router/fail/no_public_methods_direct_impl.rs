#![allow(dead_code)]
extern crate alloc;
extern crate fluentbase_sdk;
use fluentbase_sdk::{
    basic_entrypoint,
    derive::{router, Contract},
};

#[derive(Contract)]
struct App<SDK> {
    sdk: SDK,
}

// Direct implementation without using public methods
#[router(mode = "solidity")]
impl<SDK: SharedAPI> App<SDK> {
    // Private method - this should cause an error since in direct implementation
    // only public methods are included in routing
    fn private_method(&self, value: u32) -> u32 {
        value
    }

    // Another private method
    fn another_private_method(&self, message: String) -> String {
        message
    }

    // Regular deployment method - not public but special case
    fn deploy(&self) {}
}

basic_entrypoint!(App);

fn main() {}
