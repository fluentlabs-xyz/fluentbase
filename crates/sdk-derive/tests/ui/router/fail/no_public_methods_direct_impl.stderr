error: Router has no methods. Make sure your implementation contains at least one public method that is not named 'deploy'.

         = help: Check that methods are public (pub fn) for regular implementations
         = help: Consider marking your methods as public: pub fn method_name(...)

  --> tests/ui/router/fail/no_public_methods_direct_impl.rs:16:1
   |
16 | impl<SDK: SharedAPI> App<SDK> {
   | ^^^^
